### PR TITLE
feat: esett csv download - mga imbalance prices frontend

### PR DESCRIPTION
**Pull request by michaelmajgaard**

### CODE REVIEW

File: libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts [hunk 1]
```diff
@@ -183,4 +195,48 @@ export class DhMeteringGridAreaImbalanceComponent implements OnInit {
...
+
+          exportToCSVRaw({
+            content: result?.data?.downloadMeteringGridAreaImbalance,
+            fileName: 'eSett-metering-grid-area-imbalance-messages',
+          });
+
+          this.hasError = !!result.errors;
+        },
+        error: () => {
+          this.hasError = true;
```
Comment (mimse) [inline libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts:237]: Suggestion: Show a toast if an error has happened

**Review by mimse — CHANGES_REQUESTED** [review]

File: libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts [hunk 2]
```diff
@@ -183,4 +195,48 @@ export class DhMeteringGridAreaImbalanceComponent implements OnInit {
...
     this.pageMetaData$.next({ pageIndex, pageSize });
   }
+
+  download() {
+    this.isDownloading = true;
+    this.queryVariables$
+      .pipe(
+        switchMap(
+          ({ filters, documentIdSearch, sortMetadata }) =>
+            this._apollo.watchQuery({
```
Comment (dzhavat) [inline libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts:205]: suggestion: for simple request -> response action, `this._apollo.query(...)` can be used instead.

Comment (michaelmajgaard) [reply libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts]: fixed

File: libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts (hunk 1 above)
Comment (michaelmajgaard) [reply libs/dh/esett/feature-metering-gridarea-imbalance/src/lib/dh-metering-gridarea-imbalance.component.ts:238]: fixed

**Review by mimse — APPROVED** [review]

**Review by mimse — APPROVED** [review]

**Pull request merged**